pravega/pravega
# Issue 3570: Reduce recursive call to ZKOrderedStore#addEntity

Signed-off-by: tison <wander4096@gmail.com>

**Change log description**  

Flatten the recursive call to `ZKOrderedStore#addEntity` for readability.

**Purpose of the change**  

Closes #3570 

**What the code does**  

Flatten the recursive call to `ZKOrderedStore#addEntity` for readability. A known risk is that it is theoretically possible that we exhaust new collection before add the new entity but it is almost theoretically only(I think in production, we always roll over after about 1.8 billion).

**How to verify it**  

Existing tests should guard we don't change the behavior. This change itself is a trivial refactor that doesn't deserve a dedicate test.(And the only difference to current codebase is a theoretical failure that is hard to reproduce and can be safely ignored.)

cc @shiveshr @fpj
